src/timesync/timesyncd-manager.c

```diff
@@ -756,6 +756,11 @@ static int manager_resolve_handler(sd_resolve_query *q, int ret, const struct ad
                         continue;
                 }
 
+                if (ai->ai_addr->sa_family == AF_INET6 && !socket_ipv6_is_enabled()) {
+                        log_debug("Ignoring IPv6 address for %s.", m->current_server_name->string);
+                        continue;
+                }
```

Comment:
hmm, a better approach would be to change the `hints` structure that is passed into `resolve_getaddrinfo()` to set `.ai_protocol` to AF_INET4 so that the query is done for IPv4 only in the first place. i.e. a tighter query is better than filtering the answer

Reply:
(see the getaddrinfo() man page, since that's what resolve_getaddrinfo() wraps)

Reply:
You mean .ai_family, instead of .ai_protocol?

Reply:
yeah, apparently!